File: code/programming-intro.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main() {
	int addResult;
	int subtractResult;
	int multiplyResult;
	int divideResult;
	int incrementResult;
	int decrementResult;
	bool greaterThanResult;
	bool lessThanResult;
	bool equalsResult;
	bool doesNotEqualResult;

	int x = 7;
	int y = 4;
    printf("x is %d\n", x);
    printf("y is %d\n", y);
	addResult = x + y;
	subtractResult = x - y;
	multiplyResult = x * y;
	divideResult = x / y;
	greaterThanResult = x > y;
	lessThanResult = x < y;
	equalsResult = x == y;
	doesNotEqualResult = x != y;
	incrementResult = ++x;
	decrementResult = --y;

	printf("The result is x + y is %d\n", addResult);
    printf("The result is x - y is %d\n", subtractResult);
    printf("The result is x * y is %d\n", multiplyResult);
    printf("The result is x / y is %d\n", divideResult);
    printf("The result is x > y is %d\n", greaterThanResult);
    printf("The result is x < y is %d\n", lessThanResult);
    printf("The result is x == y is %d\n", equalsResult);
    printf("The result is x != y is %d\n", doesNotEqualResult);
    printf("The result is ++x is %d\n", incrementResult);
    printf("The result is --y is %d\n", decrementResult);
	printf("Operator demo complete.\n\n");

	// example if, else if, else statement in c
	x = 1;
	y = 0;
	if (x > y) {
  		printf("x is greater than y\n");
	} else if(y > x) {
  		printf("y is greater than x\n");
	} else {
  		printf("x is equal to y\n");
	}
	printf("Conditional logic demo is done running\n\n");

	// example for statement in C
	for (int x = 0; x < 3; x++) {
  		printf("x is %d\n", x);
	}
	
	printf("For loop demo is done running.\n\n");

	// example while statement in C
	x = 5;
	while (x > 0) {
  		printf("x is %d\n", x);
  		x--;
	}
	
	printf("While loop demo done running.\n");

	return 0;
}

File: code/programming-intro.py
# Starter file documenting lots of the most important features of
# the python programming language

# Declaration of a variable. Notice that in Python, variables are not typed
x = 5
print(x)
print(type(x))

y = 5.0
print(y)
print(type(y))

z = True
print(z)
print(type(z))

# Python is unique in that there is not a character data type
# Thus, a character is represented as a string that only contains one character
exampleString = "abc"
print(exampleString)
print(type(exampleString))


# Operators in Python
addResult = None
subtractResult = None
multiplyResult = None
divideResult = None
incrementResult = None
decrementResult = None
greaterThanResult = None
lessThanResult = None
equalsResult = None
doesNotEqualResult = None

x = 7
y = 4

addResult = x + y
subtractResult = x - y
multiplyResult = x * y
# Notice that most programming languages truncate all division problems
# Thus, 7/4 returns 1 since we simply truncate(remove) the decimal value
divideResult = x / y
# Unlike C, python does not have built in decrement and incremnent operators
incrementResult = x + 1
decrementResult = x - 1
greaterThanResult = x > y
lessThanResult = x < y
equalsResult = x == y
doesNotEqualResult = x != y

# These statements are a sanity check. Change x and y and make sure that the
# results make sense!
print("The result of x + y is ", addResult)
print("The result of x - y is ", subtractResult)
print("The result of x * y is ", multiplyResult)
print("The result of x / y is ", divideResult)
print("The result of x += 1 is ", incrementResult)
print("The result of x -= 1 is ", decrementResult)
print("The result of x > y is ", greaterThanResult)
print("The result of x < y is ", lessThanResult)
print("The result of x == y is ", equalsResult)
print("The result of x != y is ", doesNotEqualResult)
# A break line is added below in order to have to code output in a more
# readable format
print("\n")


# Variables for the conditional logic demo
x = 0
y = 1

# Example of an if statement in Python
if (x > y):
    print("x is greater than y")

print("If section code block is done running\n")

# Example of if else statement in Python
if (x > y):
    print("x is greater than y")
else:
    print("x is not greater than y")
    
print("If else code block is done running\n")

# Example of an if, elif, else statement in Python

if (x > y):
    print("x is greater than y")
elif (x < y):
    print("x is less than y")
else:
    print("x is equal to y")

print("If, elif, else code block is done running\n")

# Example of a for loop in python

# The range function returns a list of integers that stops just before the value entered in as an argument (in this case, the argument is 4
forLoopBegin = 1
forLoopEnd = 4
for x in range(forLoopBegin, forLoopEnd):
    print("x is ", x)

print("For loop code block is done running\n")

whileLoopCounter = 5
while (whileLoopCounter > 0):
    print("whileLoopCounter is ", whileLoopCounter)
    whileLoopCounter -= 1

print("While loop code block is done running")


